src/combinatorics/simple_graph/subgraph.lean

```diff
@@ -61,9 +61,27 @@ structure subgraph {V : Type u} (G : simple_graph V) :=
 (edge_vert : ∀ {v w : V}, adj v w → v ∈ verts)
 (symm : symmetric adj . obviously)
 
+variables {V : Type u} {W : Type v}
+
+/-- The one-vertex subgraph. -/
+@[simps]
+protected def singleton_subgraph (G : simple_graph V) (v : V) : G.subgraph :=
+{ verts := {v},
+  adj := ⊥,
+  adj_sub := by simp [-set.bot_eq_empty],
+  edge_vert := by simp [-set.bot_eq_empty] }
```

Comment:
I've noticed that I need to remove `set.bot_eq_empty` from the simp sets, otherwise it yields terms that break the `set` API. Does anyone want to look into this?

Reply:
Asked on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/set.20is.20too.20transparent).